train.py

```diff
@@ -86,32 +92,65 @@ def train_unet(
         else:
             d = data_dir
         xs, ys, ids = load_train_data(d)
-        print(f'Loaded {len(xs)} pairs of training data')
+        print('------------------------------------------------------------')
+        print(f'Loaded {len(xs)} sets of training data')
+    # if applicable, load the validation data
+    validate = False
+    if validation_dir is not None:
+        validate = True
+        v_xs, v_ys, v_ids = load_train_data(validation_dir)
+        print('------------------------------------------------------------')
+        print(f'Loaded {len(v_xs)} sets of validation data')
+    # initialise U-net
     unet = UNet()
     # load weights if applicable 
+    weights_are = 'naive'
     if weights is not None:
         unet.load_state_dict(weights)
+        weights_are = 'pretrained'
     # define the optimiser
     optimiser = optim.Adam(unet.parameters(), lr=lr)
     # define the loss function
     if loss_function == 'BCELoss':
         loss = nn.BCELoss()
+        if validate:
+            v_loss = nn.BCELoss()
     elif loss_function == 'DiceLoss':
         loss = DiceLoss()
+        if validate:
+            v_loss = DiceLoss()
+    elif loss_function == 'WeightedBCE':
+        loss = WeightedBCELoss()
+        bce_weights = loss.chan_weights
+        if validate:
+            v_loss = WeightedBCELoss()
     else:
-        raise ValueError('Valid loss options are BCELoss and DiceLoss')
+        raise ValueError('Valid loss options are BCELoss, WeightedBCE, and DiceLoss')
+    print('------------------------------------------------------------')
+    print(f'Loss function: {loss_function}')
+    if bce_weights is not None:
+        print(f'Loss function channel weights: {bce_weights}')
+    print('Optimiser: Adam')
+    print(f'Learning rate: {lr}')
     # loop over training data 
-    device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
-    with tqdm(total=epochs*len(xs), desc='unet training') as progress:
+    device_name = 'cuda' if torch.cuda.is_available() else 'cpu'
+    device = torch.device(device_name)
+    if validate:
+        no_iter = (epochs * len(xs)) + (epochs * len(v_xs))
+    else:
+        no_iter = epochs * len(xs)
+    print('------------------------------------------------------------')
+    print(f'Training {weights_are} U-net for {epochs} epochs with batch size 1 ')
+    print(f'Device: {device_name}')
+    print('------------------------------------------------------------')
+    with tqdm(total=no_iter, desc='unet training') as progress:
```

Comment:
imho you should look into custom formatting of the tqdm progress bar so you can include the live loss. That'll make the terminal output much nicer. =)